lib/src/domain/entity/event.dart

```diff
@@ -0,0 +1,42 @@
+import '../../core/util/enums.dart';
+
+class Event<T> {
```

Comment:
This is basically the same class that `DataState` but for the Presentation layer. We have to do it like this because the responsability and the knowledge of the layers should not transpass the adjacent

Reply:
Yes, but from my perspective renaming that to `UiState` would make more sense, as it's representing a state, not an event. But it's up to you!